spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/health/DatastoreHealthIndicatorConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.cloud.gcp.autoconfigure.datastore.health;
+
+import com.google.cloud.datastore.Datastore;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.autoconfigure.health.CompositeHealthIndicatorConfiguration;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTemplate;
+import org.springframework.cloud.gcp.data.datastore.repository.support.DatastoreRepositoryFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configuration for {@link DatastoreHealthIndicator}.
+ *
+ * @author Raghavan N S
+ * @author Srinivasa Meenavalli
+ */
+@Configuration
+@ConditionalOnClass({DatastoreRepositoryFactory.class, DatastoreTemplate.class, Datastore.class})
+@ConditionalOnBean({DatastoreRepositoryFactory.class, DatastoreTemplate.class, Datastore.class})
+class DatastoreHealthIndicatorConfiguration extends CompositeHealthIndicatorConfiguration<DatastoreHealthIndicator, Datastore>	{
+	@Autowired
+	Datastore datastore;
+	DatastoreHealthIndicatorConfiguration(final Datastore datastore) {
+		this.datastore = datastore;
+	}
+	@Bean
+	@ConditionalOnMissingBean(name = "datastoreHealthIndicator")
```

Comment:
Why conditional on the name "datastoreHealthIndicator" instead of the class you created `DatastoreHealthIndicator`?

Reply:
Refer Other health indicator implementations in Spring boot actuator
https://docs.spring.io/spring-boot/docs/1.4.0.RC1/api/org/springframework/boot/actuate/autoconfigure/HealthIndicatorAutoConfiguration.MongoHealthIndicatorConfiguration.html

Reply:
In my assessment I don't think simply using Spring Data Datastore should automatically trigger this background polling. Could you add a separate setting to enable it rather than making it automatic with Spring Data Datastore?

With other Spring Data modules that rely on user-hosted servers like Mongo that may make sense. But Cloud Datastore is not something the vast majority of users want to personally poll to check that it is up...